pkg/project/secrets.go

```diff
@@ -244,6 +246,10 @@ var ErrSecretNotFound = errors.New("secret not found")
 
 // Expand will expand a variable to a secret value, if no secret exists it will return an empty string
 func (e *SecretExpander) Expand(_ string, category string, name string, isFunction bool, project *Project) (string, error) {
+	if !e.cfg.GetBool(constants.UnstableConfig) && !condition.InUnitTest() {
```

Comment:
I'd prefer we avoid using `condition.InUnitTest()` if we can. I'd rather have the tests contain logic to satisfy our code instead of the other way around.

We could address this by moving the code after the config check into a separate function and testing that instead. ie. only test the "unit" we're interested in.

Reply:
I agree. I originally thought I'd have to spawn a process to run `state config set optin.unstable true`, but I found an easier way after your comment.